src/index.js

```diff
@@ -38,19 +38,16 @@ export default class Lottie extends React.Component {
     this.setAnimationControl();
   }
 
-  componentWillUpdate(nextProps /* , nextState */) {
-    /* Recreate the animation handle if the data is changed */
-    if (this.options.animationData !== nextProps.options.animationData) {
-      this.deRegisterEvents(this.props.eventListeners);
+  componentDidUpdate(prevProps) {
+    if (prevProps.options.animationData !== this.props.options.animationData) {
```

Comment:
this is slightly incorrect. It should be `this.options` here and on line 45 instead of `prevProps.options`